src/main/java/com/minecolonies/coremod/commands/ActionMenuState.java

```diff
@@ -0,0 +1,210 @@
+
+import org.jetbrains.annotations.NotNull;
+
+import com.minecolonies.coremod.colony.CitizenData;
+import com.minecolonies.coremod.colony.Colony;
+
+import net.minecraft.entity.player.EntityPlayerMP;
+
+public class ActionMenuState
+{
+    @NotNull private final ActionMenu actionMenu;
+    @NotNull private final Map<String, ArgumentState> argumentStateByActionArgumentNameMap = new HashMap<>();
+
+    private static class ArgumentState
+    {
+        private boolean valueIsSet = false;
+        @Nullable private Object value;
+    }
+
+    public ActionMenuState(@NotNull final ActionMenu actionMenu)
+    {
+        super();
+        this.actionMenu = actionMenu;
+    }
+
+    public ActionMenu getActionMenu()
+    {
+        return actionMenu;
+    }
+
+    public boolean isValueSet(@NotNull final ActionArgument actionArgument)
+    {
+        final ArgumentState argumentState = argumentStateByActionArgumentNameMap.get(actionArgument.getName());
+        if (null == argumentState)
+        {
+            return false;
+        }
+        return argumentState.valueIsSet;
+    }
+
+    public Object getValue(@NotNull final ActionArgument actionArgument)
+    {
+        final ArgumentState argumentState = argumentStateByActionArgumentNameMap.get(actionArgument.getName());
+        if (null == argumentState)
+        {
+            return null;
+        }
+        return argumentState.value;
+    }
+
+    public void setValue(@NotNull final ActionArgument actionArgument, @NotNull final Object newValue)
+    {
+        ArgumentState argumentState = argumentStateByActionArgumentNameMap.get(actionArgument.getName());
+        if (null == argumentState)
+        {
+            argumentState = new ArgumentState();
+            argumentStateByActionArgumentNameMap.put(actionArgument.getName(), argumentState);
+        }
+        argumentState.value = newValue;
+        argumentState.valueIsSet = true;
+    }
+
+    private List<ActionArgument> getAllArgumentsList()
+    {
+        return getAllArgumentsList(actionMenu.getActionArgumentList());
+    }
+
+    private List<ActionArgument> getAllArgumentsList(@NotNull final List<ActionArgument> currentActionArgumentList)
+    {
+        final List<ActionArgument> allActionArgumentsList = new ArrayList<>();
+        for (final ActionArgument actionArgument : currentActionArgumentList)
+        {
+            allActionArgumentsList.add(actionArgument);
+            if (isValueSet(actionArgument))
+            {
+                allActionArgumentsList.addAll(getAllArgumentsList(actionArgument.getActionArgumentList()));
+            }
+        }
+        return allActionArgumentsList;
+    }
+
+    public Colony getColonyForArgument(@NotNull final String argumentName)
+    {
+        for (final ActionArgument actionArgument : getAllArgumentsList())
+        {
+            if (argumentName.equals(actionArgument.getName()))
+            {
+                if (ActionArgumentType.COLONY == actionArgument.getType())
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Merge this if statement with the enclosing one. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1066)